hyperbook/book/spiele/mississippi-queen/xml.md

```diff
@@ -0,0 +1,262 @@
+
+Dieses XML-Protokoll bietet eine umfassende Struktur zur Modellierung eines hexagonalen Spielfelds mit Segmenten, die in einem übergeordneten Koordinatensystem platziert sind. Die Verwendung von [Cube-Koordinaten](https://www.redblobgames.com/grids/hexagons/#coordinates-cube) ermöglicht eine präzise Positionierung und Navigation im Spielfeld.
+
+#### Spielablauf
+
+Der erste Spieler erhält dann eine Zugaufforderung:
+
+```xml
+<room roomId="ROOM_ID">
+  <data class="moveRequest" />
+</room>
+```
+
+Worauf dieser innerhalb der gesetzten Zeitbeschränkung mit einem der folgenden Zug-Typen antwortet:
+
+##### Spielzug
+Ein Zug kann beispielhaft wie folgt aussehen:
+```xml
+<room roomId="ROOM_ID">
+  <move>
+      <actions>
+          <acceleration acc="1" />
+          <advance distance="2" />
+          <push direction="RIGHT" />
+          <turn direction="DOWN_RIGHT" />
+      </actions>
+  </move>
+</room>
+```
+
+Ein Zug besteht immer aus einer Liste aus Aktionen. Die Reihenfolge dieser Liste bestimmt auch, in welcher Reihenfolge die Aktionen ausgeführt werden. Das ist besonders wichtig zu beachten, da einige Aktionen nur erlaubt sind, wenn andere bereits ausgeführt wurden, oder Aktionen wie, die Beschleunigungsaktion, ist stets immer als erste Aktion erlaubt.
+
+##### Weiteres
+
+Nach Erhalt des Zuges sendet der Server den neuen Spielstatus an alle Spieler 
+und dem nächsten Spieler eine Zugaufforderung.
+So geht es hin und her bis zum [Spielergebnis](#spielergebnis).
+
+#### Ende der Kommunikation
+
+Die letzte Nachricht des Servers endet mit:
+
+```xml
+</protocol>
+```
+
+Danach wird die Verbindung geschlossen.
+
+### Spielergebnis
+
+Zum Spielende erhält jeder Spieler das Ergebnis. Es beginnt mit einer `definition`, die die Interpretation der Ergebnisse erklärt. Für jeden Spieler gibt es einen Eintrag in `scores`. Der darin enthaltene `score` schlüsselt sich auf in:
+
+- **cause**
+  - Beitrag des Spielers zum Spielende (`REGULAR`, `LEFT`, `RULE_VIOLATION`, `SOFT_TIMEOUT`, `HARD_TIMEOUT`).
+- **reason**
+  - Erklärung zu `cause`.
+- **part**
+  - Siegpunkte des Spielers (0 verloren, 1 unentschieden, 2 gewonnen) und weitere Punkteinträge entsprechend `definition`.
+- **winner**
+  - Gibt das Gewinner-Team an, wenn es eines gibt.
+
+Hier ein Beispiel:
+
+```xml
+<room roomId="ROOM_ID">
+  <data class="result">
+    <definition>
+      <fragment name="Siegpunkte">
+        <aggregation>SUM</aggregation>
+        <relevantForRanking>true</relevantForRanking>
+      </fragment>
+      <fragment name="∅ Punkte">
+        <aggregation>AVERAGE</aggregation>
+        <relevantForRanking>true</relevantForRanking>
+      </fragment>
+    </definition>
+    <scores>
+      <entry>
+        <player name="rad" team="ONE"/>
```

Comment:
Meintest du reds and blues, oder heißen die Teams wirklich so 😅

Reply:
Ja, das ist ein Schreibfehler 😄

Reply:
Wir sollten die Assoziierung der Farben zu den Teams nicht mehr so präsent haben, da wir die Farben ja auch in der GUI geändert haben. Warum nicht "Spieler 1" und "Spieler 2"?